add cssbundling-rails lesson (#23696)
TheOdinProject/curriculum: rails_programming/rails_sprinkles/css_bundling.md

That's all there is to it.

### Build script

When you use cssbundling it does a couple of things for you. Firstly, no matter which option you choose, it installs the necessary tools files necessary to get it running and makes sure the necessary import statements are added to your application. It then gives you instructions on the necessary build script you need to use.

The build script is used to take the css you've written and process it through the tooling you've selected. So if you choose tailwind it needs to take all those utility classes and make sure only those that you use are included to keep the tailwind file as small as possible. The build script is placed in your package.json file which is why you do need nodejs installed to use this gem. In production the build script is used to ensure all your assets are built correctly but in development you need to rebuild everytime there is a change. Because of this the gem ships with a command used to watch your files and rebuild when necessary.

> If you use either jsbundling or cssbundling you should always start by running `./bin/dev` in your terminal. This is a task provided by Rails to run not only your Rails server but also the js and css watch commands to ensure your assets are rebuilt if they need to be. Under the hood that command uses a gem called [foreman](https://github.com/ddollar/foreman) which uses a Procfile to run lots of different processes in the same terminal so it can run the Rails server as well as watching over your assets for changes.

Reviewer: Just a bit of suggested rewording, theres a lot to unpack with what the task does and how it works.

```suggestion
If you use the jsbundling or cssbundling gems, you should always start your app by running `./bin/dev` in your terminal. This task will run the Rails server, and the js and CSS watch commands to ensure your assets rebuild when needed. 

The  `./bin/dev` task uses a gem called [foreman](https://github.com/ddollar/foreman), which uses a Procfile to run lots of different processes in the same terminal, so that it can run the Rails server as well as watch over your assets for changes.
```

### Why use CSS Bundling?

> The gem isn't the only way you can integrate tools such as Bulma into your application. There are gems just for those libraries, npm packages or you can even reference the minified files directly hosted on a CDN.

Reviewer: ```suggestion
The CSS bundling gem isn't the only way to integrate tools such as Bulma into your application. There are gems just for those libraries, npm packages or you can even reference the minified files directly hosted on a CDN.
```

The README of the cssbundling-rails gem also mentions that for tailwind and dartsass there are already gems to bring those into your application and avoid having to have nodejs installed to use them.

> The main reason why this gem should be preferred is that is brings in those tools in a seamless Rails way straight into the asset pipeline, and gives you the watch command to make sure your assets get rebuilt when needed. This is extremely powerful and lets you develop in a much faster way and without having to bring in the entire library, which helps reduce those build times.

Reviewer: ```suggestion
However, the CSS bundling gem allows us to seamlessly install and get those tools working in a Rails application with minimal configuration and effort. This is extremely powerful and lets you develop much faster without having to bring in the entire library, which helps reduce those build times.

If you want more than the standard Rails way of producing stylesheets mapped to controller names, this should be the first gem you reach for when looking for something else.
```
